src/main/java/uk/gov/hmcts/reform/et/syaapi/service/ManageCaseRoleService.java

```diff
@@ -392,4 +426,132 @@ private List<CaseDetails> getCasesByCaseDetailsListAuthorizationAndCaseUserRole(
+        CaseUserAssignment caseUserAssignment = findCaseUserAssignmentByRoleAndCase(
+            role, caseDetails);
+        if (ObjectUtils.isEmpty(caseUserAssignment)) {
+            log.error("Unable to find case user role for case id: {} and case role: {}",
+                      caseDetails.getId(), CASE_USER_ROLE_CLAIMANT_SOLICITOR);
+            throw new ManageCaseRoleException(new Exception(
+                String.format(ManageCaseRoleConstants.EXCEPTION_CASE_USER_ROLES_NOT_FOUND, caseDetails.getId())));
+        }
+        CaseAssignmentUserRolesRequest caseAssignmentUserRolesRequest = ManageCaseRoleServiceUtil
+            .buildCaseUserRoleRequestByUserIdamIdCaseDetailsAndCaseRole(
+                caseUserAssignment.getUserId(),
+                caseDetails,
+                role);
+        HttpMethod httpMethod = RemoteServiceUtil.getHttpMethodByCaseUserRoleModificationType(
+            ManageCaseRoleConstants.MODIFICATION_TYPE_REVOKE);
+        restCallToModifyUserCaseRoles(caseAssignmentUserRolesRequest, httpMethod);
+    }
+
+    /**
+     * Finds a {@link CaseUserAssignment} that matches the specified case role within the given case.
+     *
+     * <p>This method retrieves all user role assignments associated with the provided case ID
+     * and searches for one that matches the specified case role. If no assignments are found
+     * for the case or the response is empty, a {@link ManageCaseRoleException} is thrown.
+     * If no matching role is found among the assignments, {@code null} is returned.</p>
+     *
+     * @param caseRole the role to match (e.g., "[APPLICANT]", "[RESPONDENT]")
+     * @param caseDetails the case from which to retrieve user assignments
+     * @return the matching {@link CaseUserAssignment}, or {@code null} if no match is found
+     * @throws IOException if an error occurs while retrieving user assignments
+     * @throws ManageCaseRoleException if no user assignments are found for the case
+     */
+    public CaseUserAssignment findCaseUserAssignmentByRoleAndCase(String caseRole, CaseDetails caseDetails)
+        throws IOException {
+        CaseUserAssignmentData caseUserAssignmentData =
+            fetchCaseUserAssignmentsByCaseId(caseDetails.getId().toString());
+        if (ObjectUtils.isEmpty(caseUserAssignmentData)
+            || CollectionUtils.isEmpty(caseUserAssignmentData.getCaseUserAssignments())) {
+            log.error("Unable to find case user roles for case id: {}", caseDetails.getId());
+            throw new ManageCaseRoleException(new Exception(
+                String.format(ManageCaseRoleConstants.EXCEPTION_CASE_USER_ROLES_NOT_FOUND, caseDetails.getId())));
+        }
+        for (CaseUserAssignment caseAssignedUserRole : caseUserAssignmentData.getCaseUserAssignments()) {
+            if (caseRole.equals(caseAssignedUserRole.getCaseRole())) {
+                return caseAssignedUserRole;
+            }
+        }
```

Comment:
If a user has shared a case to other solicitors, there can be more than one user with the CLAIMANTSOLICTOR role; this needs to collect all the users with that role as opposed to returning on the first match